### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by jc2729**
**What this PR does / why we need it**: Add types for TF2OpenAPI tool

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/164)

### CODE REVIEW

File: tools/tf2openapi/types/constants.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package types
+
+const (
+	servingMetaGraphTag    string = "serve"
```
Comment (ellis-bigelow) [inline tools/tf2openapi/types/constants.go]: Your linter should be complaining about these. 

ConstsShouldBeCapitolCamelCase

File: tools/tf2openapi/types/tfmetagraph.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+func NewTFMetaGraph(metaGraph pb.MetaGraphDef) TFMetaGraph {
+	return TFMetaGraph{}
+}
+
+func extractSigDefs(sigDefs map[string]*pb.SignatureDef) []TFSignatureDef {
+	return []TFSignatureDef{}
+}
+
+func (t *TFMetaGraph) Schema() *openapi3.Schema {
```
Comment (ellis-bigelow) [inline tools/tf2openapi/types/tfmetagraph.go]: Cool I like this interface.

File: tools/tf2openapi/types/tfmetagraph.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+type TFMetaGraph struct {
+	SignatureDefs [] TFSignatureDef
+}
+
+func NewTFMetaGraph(metaGraph pb.MetaGraphDef) TFMetaGraph {
+	return TFMetaGraph{}
+}
+
+func extractSigDefs(sigDefs map[string]*pb.SignatureDef) []TFSignatureDef {
```
Comment (ellis-bigelow) [inline tools/tf2openapi/types/tfmetagraph.go]: I imagine this private method's usefullness will become clear soon.

File: tools/tf2openapi/types/tfsavedmodel.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+func extractMetaGraphs(metaGraphs []*pb.MetaGraphDef) []TFMetaGraph {
+	return []TFMetaGraph{}
+}
+
+func (t *TFSavedModel) Schema() *openapi3.Schema {
+	return &openapi3.Schema{}
+}
+
+func ToOpenAPI(schema *openapi3.Schema) string {
```
Comment (ellis-bigelow) [inline tools/tf2openapi/types/tfsavedmodel.go]: You probably dont need both Schema and ToOpenAPI.

What about just ToOpenAPI

Comment (ellis-bigelow) [reply tools/tf2openapi/types/tfsavedmodel.go]: Actually, thinking about this, all these types should implement Schema() and the top level can execute the stringify.

File: tools/tf2openapi/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Tf2OpenAPI
+The tool enables generating OpenAPI 3.0 specifications for prediction HTTP requests from TensorFlow SavedModel files containing signature definitions (SignatureDefs) for models. One use case for the OpenAPI specification involves generating sample payloads in a UI.
```
Comment (ellis-bigelow) [inline tools/tf2openapi/README.md]: This is a run on sentence :). Happy to help you clean up the language, or you can take another pass yourself.

File: tools/tf2openapi/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+## Overview
+### Use Cases for OpenAPI
+This project is motivated by the following uses of an OpenAPI specification:
+* <b> Validate HTTP requests </b> (i.e. prediction requests) so the user can be notified of malformed inputs and correct them before running comparatively expensive computations on the model 
+* <b> Automatically generate user-friendly documentation </b> or UI with the API and display sample payloads for users so that they can format inputs properly
+* <b> Generate payloads </b> for benchmarking purposes, such as recording the response times of queries to an ML model
+
+### Project: TensorFlow-to-OpenAPI Transformer
```
Comment (ellis-bigelow) [inline tools/tf2openapi/README.md]: This is covered in the first section.

File: tools/tf2openapi/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+The outcome of this project is a TensorFlow-to-OpenAPI transformer which takes SavedModel protobuf binary files and generates an OpenAPI specification for prediction requests.
+
+### TensorFlow Compatibility
```
Comment (ellis-bigelow) [inline tools/tf2openapi/README.md]: Drop this for now?

File: tools/tf2openapi/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+TBD
+
+## Example
+TBD. Consider the TensorFlow sample in [docs/samples/tensorflow](https://github.com/kubeflow/kfserving/tree/master/docs/samples/tensorflow). The idea is that an OpenAPI specification can be generated from the model in Google Cloud Storage (gs://kfserving-samples/models/tensorflow/flowers) and be used to validate and/or generate the [sample input](https://github.com/kubeflow/kfserving/blob/master/docs/samples/tensorflow/input.json).   
+
+## Caveats
+* There is a dependency on protobufs defined by TensorFlow, e.g. [tensorflow/core/protobuf](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/protobuf). When compiling proto source files from the TensorFlow repository, ensure that the field `option go_package` is defined; otherwise, there will be missing Go files, causing problems with Go imports.
```
Comment (ellis-bigelow) [inline tools/tf2openapi/README.md]: Gotcha. What's the progress here?

File: tools/tf2openapi/README.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+* <b> Automatically generate user-friendly documentation </b> or UI with the API and display sample payloads for users so that they can format inputs properly
+* <b> Generate payloads </b> for benchmarking purposes, such as recording the response times of queries to an ML model
+
+### Project: TensorFlow-to-OpenAPI Transformer
+The outcome of this project is a TensorFlow-to-OpenAPI transformer which takes SavedModel protobuf binary files and generates an OpenAPI specification for prediction requests.
+
+### TensorFlow Compatibility
+TBD
+
+## Example
```
Comment (ellis-bigelow) [inline tools/tf2openapi/README.md]: Maybe drop this until there are isntructions to follow.

File: tools/tf2openapi/README.md (hunk 8 above)
Comment (jc2729) [reply tools/tf2openapi/README.md]: I narrowed down the necessary proto files to a few files in the Tensorflow repo, so we shouldn't need to manually add `option go_package` to the source files anymore. I'll update the README with more specific instructions on which files to compile.

File: tools/tf2openapi/README.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+# Tf2OpenAPI
+The tool enables generating OpenAPI 3.0 specifications for prediction HTTP requests from TensorFlow SavedModel files. These files contain signature definitions (SignatureDefs) for models.
```
Comment (ellis-bigelow) [inline tools/tf2openapi/README.md]: "for prediction HTTP requests from" doesn't really make sense here.

File: tools/tf2openapi/types/constants.go [hunk 11]
```diff
@@ -0,0 +1,7 @@
+package types
+
+const (
```
Comment (ellis-bigelow) [inline tools/tf2openapi/types/constants.go]: Unless these constants are used widely, consider just putting them in the file where they're used.

**Pull request merged**